File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/model/StatusDetailRoot.java
package com.ikmr.banbara23.yaeyama_liner_checker.model;

import com.ikmr.banbara23.yaeyama_liner_checker.model.time_table.TimeTable;

/**
 * ステータス詳細を表示するルートモデル
 * <p>
 * RxJavaのzipで使うために作成した
 */
public class StatusDetailRoot {
    private PortStatus mPortStatus;
    private DetailLinerInfo mDetailLinerInfo;
    private TimeTable mTimeTable;

    public StatusDetailRoot(PortStatus portStatus, DetailLinerInfo detailLinerInfo, TimeTable timeTable) {
        mPortStatus = portStatus;
        mDetailLinerInfo = detailLinerInfo;
        mTimeTable = timeTable;
    }

    public PortStatus getPortStatus() {
        return mPortStatus;
    }

    public DetailLinerInfo getDetailLinerInfo() {
        return mDetailLinerInfo;
    }

    public TimeTable getTimeTable() {
        return mTimeTable;
    }
}

File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/model/StatusDetailRoot.kt
package com.ikmr.banbara23.yaeyama_liner_checker.model

import com.ikemura.shared.model.statusdetail.PortStatus
import com.ikmr.banbara23.yaeyama_liner_checker.model.time_table.TimeTable

/**
 * ステータス詳細を表示するルートモデル
 *
 *
 * RxJavaのzipで使うために作成した
 */
data class StatusDetailRoot(
    val portStatus: PortStatus,
    val detailLinerInfo: DetailLinerInfo,
    val timeTable: TimeTable
)
